### PR TITLE
Target gutenberg monorepo branch

### PR DESCRIPTION
**Pull request by Tug**
## Description

The goal of this PR is to target the gutenberg monorepo branch. With this change `gutenberg` will receive more responsibility and gutenberg-mobile becomes a "customized experience" of the core RN editor. The idea is to have tests at both places. On gutenberg, integrated as part of the default CI and here as part of our usual contributions and releases.

Gutenberg PR: https://github.com/WordPress/gutenberg/pull/21332
WordPress-Android PR: https://github.com/wordpress-mobile/WordPress-Android/pull/11956
WordPress-iOS PR: https://github.com/wordpress-mobile/WordPress-iOS/pull/14188

## Testing instructions

Check that all the following commands are working correctly. When running the "enhanced" gutenberg mobile you should see "Welcome to gutenberg for WP apps" as the post title while it should just display "Welcome to gutenberg" for the core version.

- [x] `npm install`
- [x] `npm start`
- [x] `npm run start:reset`
- [x] `npm run core android`
- [x] `npm run core ios`
- [x] `npm run genstrings`
- [x] `npm run bundle`
- [x] `npm run test`
- [x] `npm run core test`
- [x] `npm run test:e2e:android:local`
- [x] `npm run test:e2e:ios:local`
- [x] `npm run core test:e2e:android:local`
- [x] `npm run core test:e2e:ios:local`
- [x] `npm run wpandroid`
- [x] Running the WPAndroid app in debug mode
- [x] Running the WPAndroid app in prod mode
- [x] Running the WPiOS app in debug mode
- [x] Running the WPiOS app in prod mode
- [x] CI tests running green in this PR
- [x] CI tests running green in the gutenberg PR

## Code review

The "Hide deleted files" option on github should help with the review: 

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by mchowning** [comment]
:wave: @Tug ! Let's move this to the 1.30 milestone? If this needs to make 1.29, just let me know. :bow:

File: bin/generate-podspecs.sh [hunk 1]
```diff
@@ -12,14 +12,16 @@ command -v pod > /dev/null || ( echo Cocoapods is required to generate podspecs;
 command -v jq > /dev/null || ( echo jq is required to generate podspecs; exit 1 )
 
 WD=$(pwd)
-DEST="${WD}/react-native-gutenberg-bridge/third-party-podspecs"
+DEST="${WD}/third-party-podspecs"
```
Comment (SergioEstevao) [inline bin/generate-podspecs.sh:15]: If we are keeping `generate-podspecs.sh` script on gb-mobile, shouldn't we remove(not move it) to the gutenberg repo?

We probably should also delete `gutenberg-monorepo/packages/react-native-bridge/third-party-podspecs`
folder from GB too.

**Comment by mchowning** [comment]
When I run `npm run wpandroid` and start the block editor, I'm seeing this error both with the metro bundler and with the packaged bundle. The demo app runs fine for me though. Do I just have something weird going on locally with my machine?



One other thing I noticed is that we are generating a bundle when we run `npm run wpandroid` which takes a bit of time (1-2 minutes generally). In order to speed up the build, that might be something we want to suppress with this command based on the assumption that these builds will be generally used by devs who have the metro bundler running. That's certainly not a blocker though and could be done later after these changes have been merged.

**Comment by mchowning** [comment]
`npm run core test` is working for me, but `npm run test` is giving me this error:

```
 FAIL  src/test/index.js
  ● Test suite failed to run

    Cannot find module '@wordpress/babel-plugin-import-jsx-pragma' from '/Users/matt/dev/a8c/WordPress-Android/libs/gutenberg-mobile'

      at Function.resolveSync [as sync] (node_modules/resolve/lib/sync.js:89:15)
      at resolveStandardizedName (node_modules/@babel/core/lib/config/files/plugins.js:101:31)
      at resolvePlugin (node_modules/@babel/core/lib/config/files/plugins.js:54:10)
      at loadPlugin (node_modules/@babel/core/lib/config/files/plugins.js:62:20)
      at createDescriptor (node_modules/@babel/core/lib/config/config-descriptors.js:154:9)
      at node_modules/@babel/core/lib/config/config-descriptors.js:109:50
          at Array.map (<anonymous>)
      at createDescriptors (node_modules/@babel/core/lib/config/config-descriptors.js:109:29)
      at createPluginDescriptors (node_modules/@babel/core/lib/config/config-descriptors.js:105:10)
      at plugins (node_modules/@babel/core/lib/config/config-descriptors.js:40:19)

Test Suites: 1 failed, 1 total
```

File: bin/generate-podspecs.sh (hunk 1 above)
Comment (ceyhun) [reply bin/generate-podspecs.sh:15]: Both the script and the folder are deleted, the changes are in this Gutenberg PR: https://github.com/WordPress/gutenberg/pull/21332

Comment (SergioEstevao) [reply bin/generate-podspecs.sh:15]: Thanks for the clarification.

**Comment by SergioEstevao** [comment]
Tested the following:

- [x] `npm install`
- [x] `npm start`
- [x] `npm run start:reset`
- [x] `npm run core android`
- [x] `npm run core ios`
- [x] `npm run genstrings`
- [x] `npm run bundle`
- [x] `npm run test`
- [x] `npm run core test`
- [x] `npm run test:e2e:android:local`
- [x] `npm run test:e2e:ios:local`
- [x] `npm run core test:e2e:android:local`
- [x] `npm run core test:e2e:ios:local`
- [x] `npm run wpandroid`
- [x] Running the WPAndroid app in debug mode
- [x] Running the WPAndroid app in prod mode
- [x] Running the WPiOS app in debug mode
- [x] Running the WPiOS app in prod mode
- [x] CI tests running green in this PR
- [x] CI tests running green in the gutenberg PR

**Comment by guarani** [comment]
- [x] `npm install`
- [x] `npm start`
- [x] `npm run start:reset`
- [x] `npm run core android`
- [x] `npm run core ios`
- [x] `npm run genstrings`
- [x] `npm run bundle`
- [x] `npm run test`
- [x] `npm run core test`
- ⚠️ `npm run test:e2e:android:local // some failing tests`
- ⚠️ `npm run test:e2e:ios:local // some failing tests`
- ⚠️ `npm run core test:e2e:android:local // some failing tests`
- ⚠️ `npm run core test:e2e:ios:local // all tests failing`
- ⚠️ `npm run wpandroid // error Android project not found`
- [ ] Running the WPAndroid app in debug mode
- [ ] Running the WPAndroid app in prod mode
- [x] Running the WPiOS app in debug mode
- [x] Running the WPiOS app in prod mode	// build 29070

---

#### Cleaning the project fixed some build errors

At first, I got an error when running these commands:
```
Node Sass could not find a binding for your current environment: OS X 64-bit with Node.js 12.x
```
which were resolved by cleaning the project (I'll quote here the fix that worked for me, via @Tug):

```
// In gutenberg-mobile directory
npm run clean && rm -rf node_modules
cd gutenberg && npm run clean:packages && rm -rf node_modules
// if that doesn't do it try cleaning up untracked files
git clean -dfx
// then install dependencies from gutenberg-mobile:
npm install
```

#### I saw a lot of failing tests on my machine

I saw lots of failing tests, some test suites had only a few failing tests while others had all their test fail. A common error seems to be `editorPage` being undefined – not sure why this is happening on my machine.

**Comment by mchowning** [comment]
We're preparing to cut the 1.30 release branches, so I'm going to bump this to 1.31. If this needs to make the 1.30 release though, just let me know. 🙂

**Comment by hypest** [comment]
[Pushed](https://github.com/wordpress-mobile/gutenberg-mobile/pull/2172/commits/51d42e8ceb9c14efb9e4a1ed7cfd72f460305d77) a `patch` based solution for the patching issue @cameronvoell [reported here](https://github.com/wordpress-mobile/WordPress-Android/pull/11956#issuecomment-646401082). Note that the solution won't probably be compatible with Windows as it's based on `patch`. It's a stopgap solution and we should revise with something cleaner.

**Comment by ceyhun** [comment]
@hypest The patch is printing `Skipping patch.` on CircleCI although it's a fresh `npm install` and it should be applied. I think it's using the wrong -p option. It should be `-p1` instead of `-p0`.

**Pull request merged**